### PR TITLE
feat: date filtering with calendar

### LINKED ISSUE #137: Date Filtering with calendar
#### Description 
Implement a calendar component allowing users to view and filter events by date.

#### User Story 
As a user, I want to filter events by date so that I can quickly find events that fit my schedule.

#### Acceptance Criteria 
- [ ] Date filter works as expected.
- [ ] Unit tests for filtering logic.

### PR DESCRIPTION
**Pull request by walid-as**
# Description:
## Purpose of the change
This PR implements a calendar component in the `FilterDialog` allowing users to filter events by date. It integrates `EventFilterViewModel` to manage temporary start and end dates, and supports both date range presets and custom date range selection using the Compose `DateRangePicker`. The PR also includes comprehensive UI and ViewModel tests to ensure correct behavior and full coverage of the filtering logic.

## Related issues
Closes #137
Part of #90

## How to test
* Check out the branch.
* Run the app and navigate to the `FeedScreen`.
* Tap the filter button to open the `FilterDialog`.
* Test date range presets (Today, Next Weekend, Next 7 Days).
* Test custom date range selection via the calendar dialog.
* Verify that the Confirm button only appears after both start and end dates are selected.
* Ensure the Apply Filters button is enabled when a new date range is selected.
* Apply filters and verify `ActiveFiltersBar` appears.
* Use Clear All button to reset filters.
* Run tests in `EventFilterViewModelTest` and `FilterDialogTest`.

<p align="center">
 Screenshot from 2025-11-05 23-28-35
 Screenshot from 2025-11-05 23-32-30
</p>

**Note:** An LLM was used for the sole and exclusive purpose of improving the clarity and formatting of the PR description.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -32,7 +32,7 @@ android {
 
     defaultConfig {
         applicationId = "ch.onepass.onepass"
```
Comment (M-e-d-A) [inline app/build.gradle.kts]: minSdk changed from 28→29 in build.gradle.kts. This drops Android 9 (Pie) support, affecting a non negligeable percentage of devices. The PR description doesn't mention this change. If DateRangePicker requires API 29+, please document this explicitly.

File: app/src/main/java/ch/onepass/onepass/ui/eventfilters/ActiveFiltersBar.kt [hunk 2]
```diff
@@ -92,6 +99,7 @@ fun ActiveFiltersBar(
   }
 }
 
+/** Formats a date range as a human-readable string. */
```
Comment (M-e-d-A) [inline app/src/main/java/ch/onepass/onepass/ui/eventfilters/ActiveFiltersBar.kt]: **Nitpick:** `formatDateRange()` is marked `internal` but not tested directly. Consider unit tests for edge cases (same day range, year boundaries).

File: app/src/main/java/ch/onepass/onepass/ui/eventfilters/EventFilterDialog.kt [hunk 3]
```diff
@@ -215,19 +240,74 @@ private fun FilterSection(title: String, content: @Composable () -> Unit) {
...
+        }
       },
-      confirmButton = { Button(onClick = onDismiss) { Text("OK") } },
-  )
+      dismissButton = { TextButton(onClick = onDismiss) { Text("Cancel") } })
+}
+
+/** Extends a [ClosedRange] to include the full end-of-day time for the end date. */
+private fun ClosedRange<Long>.inclusiveEndOfDay(): ClosedRange<Long> {
+  val cal = Calendar.getInstance().apply { timeInMillis = this@inclusiveEndOfDay.endInclusive }
```
Comment (M-e-d-A) [inline app/src/main/java/ch/onepass/onepass/ui/eventfilters/EventFilterDialog.kt]: **Nitpick:** Magic numbers in `inclusiveEndOfDay()`: `23`, `59`, `999`. Extract as constants: `END_OF_DAY_HOUR`, `END_OF_DAY_MINUTE`, etc.

File: app/src/main/java/ch/onepass/onepass/ui/eventfilters/EventFilterViewModel.kt [hunk 4]
```diff
@@ -7,21 +7,27 @@ import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.launch
 
+/** UI state for the filter dialog, including temporary selections. */
```
[the remark this answers is not in the document]
Comment (walid-as) [reply app/src/main/java/ch/onepass/onepass/ui/eventfilters/EventFilterViewModel.kt]: Good remark. It's fixed now :)

File: app/src/main/java/ch/onepass/onepass/ui/eventfilters/ActiveFiltersBar.kt (hunk 2 above)
Comment (walid-as) [reply app/src/main/java/ch/onepass/onepass/ui/eventfilters/ActiveFiltersBar.kt]: Fixed :)

File: app/src/androidTest/java/ch/onepass/onepass/ui/eventsfilters/EventFilterViewModelTest.kt [hunk 5]
```diff
@@ -154,4 +154,23 @@ class EventFilterViewModelTest {
     testDispatcher.scheduler.advanceUntilIdle()
     assertTrue("All filters should return true", viewModel.currentFilters.value.hasActiveFilters)
   }
+
+  @Test
+  fun tempStartAndEndDate_setAndConfirm_updatesLocalFilters() = runTest {
+    val viewModel = EventFilterViewModel()
+    val start = System.currentTimeMillis()
```
[the remark this answers is not in the document]
Comment (walid-as) [reply app/src/androidTest/java/ch/onepass/onepass/ui/eventsfilters/EventFilterViewModelTest.kt]: This test is obsolete now since I unified state handling.
